tools/lambda-promtail/lambda-promtail/s3.go

```diff
@@ -197,10 +197,11 @@ func getLabels(record events.S3EventRecord) (map[string]string, error) {
 	labels["bucket"] = record.S3.Bucket.Name
 	labels["bucket_owner"] = record.S3.Bucket.OwnerIdentity.PrincipalID
 	labels["bucket_region"] = record.AWSRegion
-	var matchingType *string
 	for key, p := range parsers {
 		if p.filenameRegex.MatchString(labels["key"]) {
-			matchingType = aws.String(key)
+			if labels["type"] == "" {
+				labels["type"] = key
+			}
```

Comment:
am I understanding the reason for moving the code here, to get rid of the chance of a panic if `matchingType` was still a nil pointer?

IMO if having a value for the `type` label is a requirement for being able to correctly parse an AWS or s3 log in `parseS3log` then we should probably also check for the existence of a `type` label explicitly rather than just erroring `there's no parser found for ""`

we could check in `processEvent` or `parseS3log`, but imo we should explicitly check for empty type


@sapslaj what do you think about that?

Reply:
@cstyan Good point. It shouldn't be too difficult to add a check and return an error in `processS3Event` if `type` isn't set, that way it won't do a GetObject for a log that will inevitably error out.

Reply:
@cstyan Alright, that's implemented! I wanted to put the check in `processS3Event` but ended up putting it in `getLabels` so it could be tested.